File: backend/src/LibraryBackend/DTOs/Responses/GetBooksResponse.cs
namespace LibraryBackend.DTOs.Responses;

public class GetBooksResponse
{
    public int Id { get; init; }

    public string Title { get; init; } = null!;

    public string Cover { get; init; } = null!;

    public string Author { get; init; } = null!;

    public double Rating { get; init; }

    public int ReviewsNumber { get; init; }
}

File: frontend/src/modules/book/dto/responses/get-books-response.ts
export class GetBooksResponse {
  id: number;
  title: string;
  cover: string;
  author: string;
  rating: number;
  reviewsNumber: number;

  constructor(
    id: number,
    title: string,
    cover: string,
    author: string,
    rating: number,
    reviewsNumber: number) {

    this.id = id;
    this.title = title;
    this.cover = cover;
    this.author = author;
    this.rating = rating;
    this.reviewsNumber = reviewsNumber;
  }
}
